tests/e2e/matmul/BUILD

```diff
@@ -128,6 +128,11 @@ py_binary(
     target_backends_and_drivers = [
         ("vmvx", "local-task"),
     ],
+    target_cpu_features_variants = ["default"] +
+                                   ([
+                                       "aarch64:+dotprod",
```

Comment:
We're getting to pretty complex logic in the build files here. I'm not convinced that the list comprehension is helping us at this point. This generates 4 rules and actually all 4 of them take different branches on these various if conditions. I think we should consider expanding them out and possibly adding one or more macros to capture the meaningful similarities

Reply:
OK, split the mmt4d vs direct cases (ie dropped the `for` on `strategy`). Initially I also split by lhs_rhs_type (dropped the other `for`) but found that that one made more sense to keep. It's a trade-off with verbosity: long repetitive BUILD code with subtle variations isn't easy to read either, and the list comprehension trick serves a purpose, it factors out the repetitiveness and highlights the variations. WDYT?

Reply:
Yeah it's definitely a balance. One of the arguments inside Google around not doing this sort of thing is that it makes it harder for tools to process (e.g. buildozer), but I think that's less relevant for us. This seems fine to me